tinkerpop3/src/main/java/io/shiftleft/overflowdb/storage/NodeDeserializer.java

```diff
@@ -22,31 +22,25 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
-public class NodeDeserializer {
-  private final Logger logger = LoggerFactory.getLogger(getClass());
+public class NodeDeserializer extends BookKeeper {
   protected final OdbGraph graph;
   private final Map<Integer, NodeFactory> nodeFactoryByLabelId;
-  private int deserializedCount = 0;
-  private long deserializationTimeSpentNanos = 0;
   private ConcurrentHashMap<String, String> interner;
 
-
-  public NodeDeserializer(OdbGraph graph, Map<Integer, NodeFactory> nodeFactoryByLabelId) {
+  public NodeDeserializer(OdbGraph graph, Map<Integer, NodeFactory> nodeFactoryByLabelId, boolean statsEnabled) {
+    super(statsEnabled);
     this.graph = graph;
     this.nodeFactoryByLabelId = nodeFactoryByLabelId;
-    this.interner = new ConcurrentHashMap<String, String>();
+    this.interner = new ConcurrentHashMap<>();
   }
 
   private final String intern(String s){
     String interned = interner.putIfAbsent(s, s);
     return interned == null ? s : interned;
   }
 
-
   public final OdbNode deserialize(byte[] bytes) throws IOException {
-    // todo: only time when some config is set
-
-    long start = System.nanoTime();
+    long startTimeNanos = System.nanoTime();
```

Comment:
We would probably need `long startTimeNanos = statsEnabled ? System.nanoTime() : 0;`.

The point is that `System.nanoTime()` is afaiu a syscall on linux+openjdk8 (relatively fast, but this depends a lot on system config). This means that it might be surprisingly expensive, depending on setup. We absolutely want to prevent a setting where the message deserialization takes 300 ns, and we pay an extra 400 ns for a roundtrip to the kernel, just to collect stats we don't even collect. 

If this was an `RDTSC`, I wouldn't care (24 cycles on broadwell); but `System.nanoTime()` makes a lot of expensive extra guarantees.

Reply:
good point, yup makes sense